src/snowflake/snowpark/mock/_pandas_util.py

```diff
@@ -202,17 +207,23 @@ def convert_to_python_obj(obj):
     for idx, pandas_type in enumerate(data.dtypes):
         if isinstance(pandas_type, pd.IntervalDtype):
             data_type = VariantType()
-        elif isinstance(pandas_type, pd.DatetimeTZDtype):
-            data_type = TimestampType()
+        elif (
+            isinstance(pandas_type, pd.DatetimeTZDtype)
+            or pandas_type.type == numpy.datetime64
+        ):
+            if isinstance(pandas_type, pd.DatetimeTZDtype) and pandas_type.tz:
+                LocalTestOOBTelemetryService.get_instance().log_not_supported_error(
+                    external_feature_name="DataFrame creation from pandas DataFrame containg pd.DatetimeTZDtype with timezone information",
+                    internal_feature_name="_pandas_util._extract_schema_and_data_from_pandas_df",
+                    raise_error=NotImplementedError,
+                )
+            else:
+                data_type = TimestampType(TimestampTimeZone.NTZ)
         elif pandas_type.type == numpy.float64:
             data_type = DoubleType()
         elif isinstance(pandas_type, (pd.Float32Dtype, pd.Float64Dtype)):
             data_type = DoubleType()
-        elif (
-            pandas_type.type == numpy.int64
-            or pandas_type.type == numpy.datetime64
```

Comment:
Is datetime64 covered by a different branch now?

Reply:
this, this is moved to line 210 where DateTime is handled